internal/run/workflow.go

```diff
@@ -272,3 +273,11 @@ func (w *Workflow) Clone(ctx context.Context, opts ...Opt) (*Workflow, error) {
 		)...,
 	)
 }
+
+func Migrate(ctx context.Context, wf *workflow.Workflow) {
+	if registry.IsRegistryEnabled(ctx) {
```

Comment:
Looks like this check includes telemetryDisabled := core.IsTelemetryDisabled(ctx) all good